### PR TITLE
Infer joined and active member count from the member list, default t…

### PR DESCRIPTION
**Pull request by stefanceriu**
…he members to all mock members

### CODE REVIEW

File: ElementX/Sources/Mocks/RoomMemberProxyMock.swift [hunk 1]
```diff
@@ -103,3 +111,15 @@ extension RoomMemberProxyMock {
                                         canSendStateEvent: { allowedStateEvents.contains($0) }))
     }
 }
+
+extension Array where Element == RoomMemberProxyMock {
+    static let allMembers: [RoomMemberProxyMock] = [
```
Comment (pixlwave) [inline ElementX/Sources/Mocks/RoomMemberProxyMock.swift:116]: Nice 👌

File: UnitTests/__Snapshots__/PreviewTests/test_roomDetailsScreen.DM-Room.png
Comment (pixlwave) [inline UnitTests/__Snapshots__/PreviewTests/test_roomDetailsScreen.DM-Room.png:1]: It's not an issue with this PR, but is this test actually configured correctly? Looks like it's showing room details and not the alternative DM details configuration 🤔

File: ElementX/Sources/Mocks/RoomProxyMock.swift [hunk 2]
```diff
@@ -66,15 +63,13 @@ extension RoomProxyMock {
...
+        members = CurrentValueSubject(configuration.members).asCurrentValuePublisher()
 
-        if let configuredMembers = configuration.members {
-            members = CurrentValueSubject(configuredMembers).asCurrentValuePublisher()
-        } else {
-            members = CurrentValueSubject([]).asCurrentValuePublisher()
-        }
+        let joinedMembers = configuration.members.filter { $0.membership == .join }
+        joinedMembersCount = joinedMembers.count
+        activeMembersCount = joinedMembers.count
```
Comment (pixlwave) [inline ElementX/Sources/Mocks/RoomProxyMock.swift:72]: To be accurate shouldn't this filter to include `.join` and `.invited`? (I think that's the definition of active right).

**Review by pixlwave — APPROVED** [review]
Nice tidy-up here 👍

File: ElementX/Sources/Mocks/RoomProxyMock.swift (hunk 2 above)
Comment (stefanceriu) [reply ElementX/Sources/Mocks/RoomProxyMock.swift:72]: Oh, I didn't know that 😅 I thought it's members that have been active recently

**Pull request merged**